scripts/reindex.py

```diff
@@ -98,18 +108,52 @@ def main(argv: List[str]):
 
     azul = AzulClient(num_workers=args.num_workers)
 
+    source_globs = set(args.sources)
+    if args.partition_prefix_length:
+        sources_by_catalog = defaultdict(set)
+        globs_matched = set()
+        for catalog in args.catalogs:
+            sources = azul.catalog_sources(catalog)
+            for source_glob in source_globs:
+                matches = fnmatch.filter(sources, source_glob)
+                if matches:
+                    globs_matched.add(source_glob)
+                logger.debug('Source glob %r matched sources %r in catalog %r',
+                             source_glob, matches, catalog)
+                sources_by_catalog[catalog].update(matches)
+        unmatched = source_globs - globs_matched
+        if unmatched:
+            logger.warning('Source(s) not found in any catalog: %r', unmatched)
+        require(any(sources_by_catalog.values()),
+                'No valid sources specified for any catalog')
+    else:
+        if source_globs == {'*'}:
+            sources_by_catalog = {
```

Comment:
This looks like it creates a set of dictionaries. That can't be right. Was this branch of the control flow tested?

Reply:
Seems I overlooked the case where `*` is used without any additional sources.

Reply:
I misinterpreted the context here, I did test `*` by itself, it was the local reindex I overlooked.